src/middleware.rs

```diff
@@ -280,6 +299,36 @@ impl StatementArg {
     }
 }
 
+impl ToFields for StatementArg {
+    fn to_fields(self) -> (Vec<F>, usize) {
+        // NOTE: current version returns always the same amount of field elements in the returned
```

Comment:
Actually I think always returning the same amount of field elements is much safer, even in the future, because if we use a dynamic length encoding then we may encounter more instances of collisions (different lists of StatementArg that are translated to the same vector of fields)